db/migrate/20170120164058_create_task_for_each_job_and_transfer_attributes.rb

```diff
@@ -8,7 +8,28 @@ class MiqTask < ActiveRecord::Base
     has_one :job, :class_name => 'CreateTaskForEachJobAndTransferAttributes::Job'
   end
 
+  class LogFile < ActiveRecord::Base
+  end
+
+  class BinaryBlob < ActiveRecord::Base
+  end
+
   def up
+    say_with_time("Deleting finished jobs older than 7 days") do
+      Job.where("updated_on < ?", 7.days.ago.utc).where("state = 'finished'").delete_all
+    end
+
+    say_with_time("Deleting logs linked to finished tasks older than 7 days") do
+      LogFile.where(:miq_task_id => MiqTask.where("updated_on < ?", 7.days.ago.utc).select(:id)).delete_all
+    end
+    say_with_time("Deleting BinaryBlobs linked to finished tasks older than 7 days") do
+      BinaryBlob.where(:resource_type => 'MiqTask', :resource_id => MiqTask.where("updated_on < ?", 7.days.ago.utc).select(:id)).delete_all
+    end
+
+    say_with_time("Deleting finished tasks older than 7 days") do
+      MiqTask.where("updated_on < ?", 7.days.ago.utc).where("state = 'Finished'").delete_all
```

Comment:
Minor, but I would move that out to a variable at the top.  If any of these actions take some time, then it's a moving window, which is something we don't want as it could create orphans.

Reply:
I think the `.where("state = 'Finished'")` needs to apply to the conditionals above for LogFile and BinaryBlob.  Perhaps extracts the query fragment out to a variable and reuse it.